.github/workflows/docker-build-and-push-arm64.yaml

```diff
@@ -61,6 +61,39 @@ jobs:
             *.cache-from=type=registry,ref=ghcr.io/${{ github.repository }}-buildcache:arm64-main
             *.cache-to=type=registry,ref=ghcr.io/${{ github.repository }}-buildcache:arm64-${{ github.ref_name }},mode=max
 
+      - name: Show disk space
+        if: always()
+        run: |
+          df -h
+
+  docker-build-and-push-cuda:
```

Comment:
I think we once integrated no-cuda and cuda jobs to quicken the build since we only have one arm instance. Do we want to split the jobs again? (Sorry if it was already discussed in an Open AD Kit WG call)

Reply:
It's a good question. In https://github.com/autowarefoundation/autoware/pull/5363, we unified the jobs into one. However, it turned out that even after running `free disk space` action, docker images, docker build caches, and so on remain, causing the `docker-build-and-push` workflow to fail due to disk errors sometimes https://github.com/autowarefoundation/autoware/actions/runs/11779518985.

Therefore, it became clear that we still need to split the jobs into two. However, unlike the previous parallel workflow, this PR sets up a sequential workflow. Additionally, since the latter job only handles CUDA packages, the total runtime should be shorter than before.

- Before
<img width="649" alt="Screenshot 2024-11-12 at 10 54 25" src="https://github.com/user-attachments/assets/d8515b9d-0422-4a1c-91fe-cb8c4b9d7094">

- After
<img width="945" alt="Screenshot 2024-11-12 at 10 51 29" src="https://github.com/user-attachments/assets/15a34e97-ea98-4914-b17c-498ff950925e">